connector/apmconnector/metric_to_metric_connector.go

```diff
@@ -261,6 +275,92 @@ func createResourceAndScopeMetrics(logger *zap.Logger, rmNew pmetric.ResourceMet
 	return rmNew, smNew, metrics
 }
 
+func convertUnitsHistogramDataPoint(dp pmetric.HistogramDataPoint, conversionFactor float64) {
+	if conversionFactor == 1 {
+		return
+	}
+
+	if dp.HasSum() {
+		dp.SetSum(dp.Sum() * conversionFactor)
+	}
+	if dp.HasMin() {
+		dp.SetMin(dp.Min() * conversionFactor)
+	}
+	if dp.HasMax() {
+		dp.SetMax(dp.Max() * conversionFactor)
+	}
+	for index, bound := range dp.ExplicitBounds().AsRaw() {
+		dp.ExplicitBounds().SetAt(index, bound*conversionFactor)
+	}
+}
+
+func convertUnitsExponentialHistogramDataPoint(dp pmetric.ExponentialHistogramDataPoint, conversionFactor float64) {
```

Comment:
I think we should test this function

Reply:
Added a simple test that shows how this method works and illustrates the precision issues that arise from conversion.

For example, take note of the bucket at index 5. In the original histogram the bucket contains 4 values that represent the range ~2 to 2.8 milliseconds.

```go
dp.Positive().BucketCounts().SetAt(5, 4)  // (2.000000, 2.828427]
```

When converted to seconds. The bucket at index 5 still has 4 values, but the values represent the range ~0.001953 to 0.002762 seconds.

```go
assert.Equal(t, uint64(4), dp.Positive().BucketCounts().At(5))  // (0.001953, 0.002762]
```

So, no matter how you do it, unit conversion for exponential histograms is lossy. We could improve a bit on the lossiness by spreading the bucket count proportionally across the corresponding buckets of the converted histogram. OTLP ingest does something similar to this when converting from OTLP format to NR's exponential histogram format. For the purposes of this prototype, I think the simple thing this PR does is good enough.